assets/js/base/hooks/use-store-cart-item.js

```diff
@@ -22,32 +24,43 @@ import { useStoreCart } from './use-store-cart';
  */
 export const useStoreCartItem = ( cartItemKey ) => {
 	const { cartItems, cartIsLoading } = useStoreCart();
-	const cartItem = cartItems.filter( ( item ) => item.key === cartItemKey );
+	const cartItem = cartItems.find( ( item ) => item.key === cartItemKey );
 
-	const results = useSelect(
-		( select, { dispatch } ) => {
+	// Store quantity in hook state. This is used to keep the UI
+	// updated while server request is updated.
+	const [ quantity, changeQuantity ] = useState( cartItem?.quantity );
```

Comment:
I had to add the careful `cartItem?.quantity` optional chain operator here because this hook fires in the editor, causing JS error. I think this is because the block is using preview data but the hook still loads real cart data.

I think we need to fix this issue (as follow up) - the cart block shouldn't pull real data in the editor at all. Not sure of the best way to do this (e.g. `shouldSelect`, `isEditor` props, or maybe we can set "isEditor" in context somewhere so hooks can transparently support this. cc @nerrad

Reply:
Ya this is because `cartItems` could change. If you implement my suggestion for deriving `cartItem` via `useEffect` then that should take care of this issue.